src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,719 @@
+     * the check once per packet: The RTCP transformer is only supposed to
+     * (reverse) transform RTCP packets and the RTP transformer is only
+     * supposed to modify RTP packets.
+     */
+    public RemoteBitrateEstimatorAbsSendTime() {
+        super();
+    }
+
+    /**
+     * Ctor.
+     *
+     * @param packetPredicate the <tt>PacketPredicate</tt> to use to match
+     *                        packets to (reverse) transform.
+     */
+    public RemoteBitrateEstimatorAbsSendTime(Predicate<ByteArrayBuffer> packetPredicate) {
+        super(packetPredicate);
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * The (default) implementation of {@code SinglePacketTransformer} does
+     * nothing.
+     */
+    @Override
+    public void close() {
+        super.close();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Reverse-transforms an array of packets by calling
+     * {@link #reverseTransform(RawPacket)} on each one.
+     *
+     * @param pkts
+     */
+    @Override
+    public RawPacket[] reverseTransform(RawPacket[] pkts) {
+        return super.reverseTransform(pkts);
+    }
+
+    /**
+     * Transforms a specific packet.
+     *
+     * @param pkt the packet to be transformed.
+     * @return the transformed packet.
+     */
+    @Override
+    public RawPacket transform(RawPacket pkt) {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Transforms an array of packets by calling {@link #transform(RawPacket)}
+     * on each one.
+     *
+     * @param pkts
+     */
+    @Override
+    public RawPacket[] transform(RawPacket[] pkts) {
+        return super.transform(pkts);
+    }
+
+    /**
+     * Returns the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}. The method is called on the same
+     * worker thread as Process will be called on.
+     *
+     * @return the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}
+     */
+    @Override
+    public long getTimeUntilNextRun() {
+        return 0;
+    }
+
+    private enum ProbeResult
+    {
+        kBitrateUpdated(0),
+        kNoUpdate(1);
+        int value;
+        ProbeResult(int x)
+        {
+            this.value = x;
+        };
+    }
+
+    private final Object critSect = new Object();
+    private Collection<Integer> ssrcs = new ArrayList<Integer>();
+    private final static int kTimestampGroupLengthMs = 5;
+    private final static int kAbsSendTimeFraction = 18;
+    private final static int kAbsSendTimeInterArrivalUpshift = 8;
+    private final static int kInterArrivalShift
+            =  kAbsSendTimeFraction + kAbsSendTimeInterArrivalUpshift;
+    private final static int kInitialProbingIntervalMs = 2000;
+    private final static int kMinClusterSize = 4;
+    private final static int kMaxProbePackets = 15;
+    private final static int kExpectedNumberOfProbes = 3;
```

Comment:
Please move all static fields to the top of the file. The order is: statics, instance fields, constructors, methods, classes.